### PR TITLE
Address inconsistency user settings on Mac/Linux

### PR DESCRIPTION
**Pull request by heng-liu**
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/2822

Update [Common NuGet configurations](https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior), explaining the inconsistency and providing the workaround of manually consolidating the two paths into one by setting up symbolic link.
Add the link of inconsistency in all the NuGet.exe commands which changes the user-wide NuGet.Config file:
[NuGet.exe config](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-config) command
[NuGet.exe sources](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-sources) command
[NuGet.exe setapikey](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-setapikey) command

### CODE REVIEW

**Review by zivkan — APPROVED** [review]

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 1]
```diff
@@ -16,9 +16,26 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
+| User | **Windows:** `%appdata%\NuGet\NuGet.Config`<br/>**Mac/Linux:** `~/.config/NuGet/NuGet.Config` or `~/.nuget/NuGet/NuGet.Config` (varies by tooling) <br/>Additional configs are supported on all platforms. These configs cannot be edited by the tooling. </br> **Windows:** `%appdata%\NuGet\config\*.Config` <br/>**Mac/Linux:** `~/.config/NuGet/config/*.config` or `~/.nuget/config/*.config` | Settings apply to all operations, but are overridden by any project-level settings. |
 | Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. If `$XDG_DATA_HOME` is null or empty, `~/.local/share` or `/usr/local/share` will be used (varies by OS distribution)  | Settings apply to all operations on the computer, but are overridden by any user- or project-level settings. |
 
+> [!Note]
+> On Mac/Linux, the user config file location varies by tooling. Dotnet CLI uses `~/.nuget/NuGet` folder, while Mono uses `~/.config/NuGet` folder.
```
Comment (JonDouglas) [inline docs/consume-packages/configuring-nuget-behavior.md:23]:
```suggestion
> On Mac/Linux, the user config file location varies by tooling. .NET CLI uses `~/.nuget/NuGet` folder, while Mono uses `~/.config/NuGet` folder. 
```

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 2]
```diff
@@ -16,9 +16,26 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
+| User | **Windows:** `%appdata%\NuGet\NuGet.Config`<br/>**Mac/Linux:** `~/.config/NuGet/NuGet.Config` or `~/.nuget/NuGet/NuGet.Config` (varies by tooling) <br/>Additional configs are supported on all platforms. These configs cannot be edited by the tooling. </br> **Windows:** `%appdata%\NuGet\config\*.Config` <br/>**Mac/Linux:** `~/.config/NuGet/config/*.config` or `~/.nuget/config/*.config` | Settings apply to all operations, but are overridden by any project-level settings. |
 | Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. If `$XDG_DATA_HOME` is null or empty, `~/.local/share` or `/usr/local/share` will be used (varies by OS distribution)  | Settings apply to all operations on the computer, but are overridden by any user- or project-level settings. |
 
+> [!Note]
+> On Mac/Linux, the user config file location varies by tooling. Dotnet CLI uses `~/.nuget/NuGet` folder, while Mono uses `~/.config/NuGet` folder. 
+
+### On Mac/Linux, the user-level config file location varies by tooling. 
+On Mac/Linux, the user config file location varies by tooling.
```
Comment (JonDouglas) [inline docs/consume-packages/configuring-nuget-behavior.md:26]:
```suggestion
```

**Review by JonDouglas — APPROVED** [review]
Looks good! Added some suggestions and we can merge!

File: docs/consume-packages/configuring-nuget-behavior.md (hunk 2 above)
Comment (heng-liu) [reply docs/consume-packages/configuring-nuget-behavior.md:26]: Thanks for reviewing! Is there a way to reference this part if it's not a heading level? As NuGet.exe command needs to add a reference to this part.

Comment (zivkan) [reply docs/consume-packages/configuring-nuget-behavior.md:26]: the suggested change is not on the heading, it's on the line after the heading that repeats what the heading already says

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 3]
```diff
@@ -16,9 +16,26 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
 | Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. If `$XDG_DATA_HOME` is null or empty, `~/.local/share` or `/usr/local/share` will be used (varies by OS distribution)  | Settings apply to all operations on the computer, but are overridden by any user- or project-level settings. |
 
+> [!Note]
+> On Mac/Linux, the user config file location varies by tooling. .NET CLI uses `~/.nuget/NuGet` folder, while Mono uses `~/.config/NuGet` folder. 
+
+### On Mac/Linux, the user-level config file location varies by tooling. 
+On Mac/Linux, the user config file location varies by tooling. 
+Majority users uses tools looking for user config file under `~/.nuget/NuGet` folder.
```
Comment (JonDouglas) [inline docs/consume-packages/configuring-nuget-behavior.md:27]:
```suggestion
Majority of users use tools that look for the user config file under the `~/.nuget/NuGet` folder. 
```

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 4]
```diff
@@ -16,9 +16,26 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
 | Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. If `$XDG_DATA_HOME` is null or empty, `~/.local/share` or `/usr/local/share` will be used (varies by OS distribution)  | Settings apply to all operations on the computer, but are overridden by any user- or project-level settings. |
 
+> [!Note]
+> On Mac/Linux, the user config file location varies by tooling. .NET CLI uses `~/.nuget/NuGet` folder, while Mono uses `~/.config/NuGet` folder. 
+
+### On Mac/Linux, the user-level config file location varies by tooling. 
+On Mac/Linux, the user config file location varies by tooling. 
+Majority users uses tools looking for user config file under `~/.nuget/NuGet` folder. 
+While following tools look for user config files under `~/.config/NuGet` folder:
```
Comment (JonDouglas) [inline docs/consume-packages/configuring-nuget-behavior.md:28]:
```suggestion
These other tools look for the user config file under the `~/.config/NuGet` folder:
```

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 5]
```diff
@@ -16,9 +16,26 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
+* Mono
+* NuGet.exe
+* Visual Studio 2019 for Mac (and earlier versions)
+* Visual Studio 2022 for Mac (and later versions), only when working on classic Mono projects.
+
+If the tooling you use involves both locations, consolidating them by following steps allows you to work with only one user-level config file:
```
Comment (JonDouglas) [inline docs/consume-packages/configuring-nuget-behavior.md:34]:
```suggestion
If the tooling you use involves both locations, consider consolidating them by following these steps to allow you to work with only one user-level config file:  
```

File: docs/consume-packages/configuring-nuget-behavior.md (hunk 2 above)
Comment (JonDouglas) [reply docs/consume-packages/configuring-nuget-behavior.md:26]: The header is great and I just removed the duplicate line.

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 6]
```diff
@@ -16,9 +16,26 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
+| User | **Windows:** `%appdata%\NuGet\NuGet.Config`<br/>**Mac/Linux:** `~/.config/NuGet/NuGet.Config` or `~/.nuget/NuGet/NuGet.Config` (varies by tooling) <br/>Additional configs are supported on all platforms. These configs cannot be edited by the tooling. </br> **Windows:** `%appdata%\NuGet\config\*.Config` <br/>**Mac/Linux:** `~/.config/NuGet/config/*.config` or `~/.nuget/config/*.config` | Settings apply to all operations, but are overridden by any project-level settings. |
 | Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. If `$XDG_DATA_HOME` is null or empty, `~/.local/share` or `/usr/local/share` will be used (varies by OS distribution)  | Settings apply to all operations on the computer, but are overridden by any user- or project-level settings. |
 
+> [!Note]
+> On Mac/Linux, the user config file location varies by tooling. .NET CLI uses `~/.nuget/NuGet` folder, while Mono uses `~/.config/NuGet` folder. 
+
+### On Mac/Linux, the user-level config file location varies by tooling.
```
Comment (nkolev92) [inline docs/consume-packages/configuring-nuget-behavior.md:25]:
```suggestion
### On Mac/Linux, the user-level config file location varies by tooling
```

**Review by nkolev92 — APPROVED** [review]

**Comment by heng-liu** [comment]
Hi @JonDouglas @nkolev92 , may I know if this is good to merge? Thanks!

**Comment by nkolev92** [comment]
🚢 🚢 🚢 🚢

**Pull request merged**